src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -1127,6 +1134,8 @@ impl_display_for_with_option!(CreateSubsourceOption);
 pub struct CreateSubsourceStatement<T: AstInfo> {
     pub name: UnresolvedItemName,
     pub columns: Vec<ColumnDef<T>>,
+    // Tracks the primary source of this subsource if a data-bearing subsource.
+    pub of_source: Option<T::ItemName>,
```

Comment:
There are two types of subsources––data bearing and the progress subsources. In my mind, the progress collection isn't "data bearing" because it's just metadata...but this term could use some workshopping.